tensorflow_datasets/image/imagenette.py

```diff
@@ -0,0 +1,139 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Imagenette: a subset of 10 easily classified classes from Imagenet (tench,
+English springer, cassette player, chain saw, church, French horn, garbage
+truck, gas pump, golf ball, parachute)"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import io
+import itertools
+import os
+
+import numpy as np
+import tensorflow as tf
+
+import tensorflow_datasets.public_api as tfds
+
+# TODO(imagenette): BibTeX citation
+_CITATION = """
+"""
+
+_DESCRIPTION = """\
+Imagenette is a subset of 10 easily classified classes from the Imagenet
+dataset. It was originally prepared by Jeremy Howard of FastAI. The objective
+behind putting together a small version of the Imagenet dataset was mainly
+because running new ideas/algorithms/experiments on the whole Imagenet take a
+lot of time.
+
+This version of the dataset allows researchers/practitioners to quickly try out
+ideas and share with others. The dataset comes in three variants:
+
+  * Full size
+  * 320 px
+  * 160 px
+This dataset consists of the Imagenette dataset {size} variant.
+"""
+
+_LABELS_FNAME = 'image/imagenette_labels.txt'
+_URL_PREFIX = 'https://s3.amazonaws.com/fast-ai-imageclas'
+
+
+class ImagenetteConfig(tfds.core.BuilderConfig):
+  """BuilderConfig for Imagenette."""
+
+  def __init__(self, size, **kwargs):
+    super(ImagenetteConfig, self).__init__(
+        version=tfds.core.Version('0.1.0'), **kwargs)
+    self.size = size
+
+
+def _make_builder_configs():
+  configs = []
+  for size in ['full-size', '320 px', '160 px']:
+    configs.append(
+        ImagenetteConfig(
+            name='%s' % (size),
+            size=size,
+            description=_DESCRIPTION.format(size=size)))
+  return configs
+
+
+class Imagenette(tfds.core.GeneratorBasedBuilder):
+  """Imagenette:A smaller subset of 10 easily classified classes from
+  Imagenet."""
+
+  VERSION = tfds.core.Version('0.1.0')
+  BUILDER_CONFIGS = _make_builder_configs()
+
+  def _info(self):
+    names_file = tfds.core.get_tfds_path(_LABELS_FNAME)
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            'image': tfds.features.Image(),
+            'label': tfds.features.ClassLabel(names_file=names_file)
+        }),
+        supervised_keys=('image', 'label'),
+        homepage='https://github.com/fastai/imagenette',
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    size = self.builder_config.size
+
+    if size in ["320 px","160 px"]:
```

Comment:
rather than repeating these lines 3 times, you could compute the "file-prefix" instead.

(file_prefix == "imagenette-160" or "imagenette-320" or "imagenette")

and then use it in the lines below.

Reply:
Good idea. Will do it.